### ISSUE TITLE
Cutoff frequency of dsp::TRCFilter, missing 2*Pi factor?

### ISSUE THREAD

**Issue by MarcBoule** [VCVRack/Rack#1289]
I'm wondering if there could be a missing 2*Pi factor in `dsp::TRCFilter.setCutoff()`.

I'm testing in v0.6 in order to use BogAudio's spectrum analyser, but I copied the new v1 code for dsp::TRCFilter into my module for good measure (Branes is temporarily modified to be a test rig for the filters). I'm feeding the filter some white noise, and I have the cutoff set like this:

`rc.setCutoff(6280.0f / engineGetSampleRate())`

Below is the frequency response (blue curve) of the RCFilter, where we can roughly see that the cutoff is at about 1000 Hz instead of the 6280 Hz that was specified above. The red curve is a [OnePole filter](http://www.earlevel.com/main/2012/12/15/a-one-pole-filter/), with its cutoff set at 1000 Hz.

`onePole.setCutoff(1000.0f / engineGetSampleRate())`

![image](https://user-images.githubusercontent.com/33604154/58107171-5235cd00-7bb7-11e9-94c1-b02774fffd67.png)

I'm still cutting my teeth on dsp, so perhaps I'm overlooking something.

**Comment by MarcBoule** [comment]
After researching it [a bit more](https://en.wikipedia.org/wiki/RC_circuit#Frequency-domain_considerations), and studying a [matlab example](https://www.mathworks.com/matlabcentral/fileexchange/24617-digital-resistor-capacitor-rc-filter?s_tid=FX_rc1_behav), i think that there indeed is a missing 2*Pi factor in `setCutoff()`.

I believe it should be (before simplification):

```c++
	void setCutoff(T r) {
		c = 2.f / (2.f * M_PI * r);
	}
```

As opposed to the current:

```c++
	void setCutoff(T r) {
		c = 2.f / r;
	}
```

**Comment by MarcBoule** [comment]
For those wondering why the rolloff in the RC filter (red curve) is quite abrupt at high frequencies and different than the one pole filter, it is because the coefficient related to the cutoff should _ideally_ be 

`A = 2 / tan(2 * Pi * fc / fs)`

but in the RC filter the `tan()` was likely removed for optimization reasons, since for small `fc/fs`, we have approximately `tan(x) = x`.

**Comment by AndrewBelt** [comment]
Taking your word for it, since I can't find a pen. Fixed in e667c7d by rewriting the documentation and adding a new method with the correct formula.
